graphql/util.go

```diff
@@ -1032,3 +1033,44 @@ func getBaseTaskTestResultsOptions(ctx context.Context, dbTask *task.Task) ([]te
 
 	return taskOpts, nil
 }
+
+func handleOnSaveOperation(ctx context.Context, distroID string, userID string, onSave DistroOnSaveOperation) (int, error) {
+	if onSave == DistroOnSaveOperationNone {
+		return 0, nil
+	}
+
+	hosts, err := host.Find(db.Query(host.ByDistroIDs(distroID)))
+	if err != nil {
+		return 0, InternalServerError.Send(ctx, fmt.Sprintf("finding hosts for distro with id '%s': %s", distroID, err))
+	}
+
+	switch onSave {
+	case DistroOnSaveOperationDecommission:
+		if err = host.DecommissionHostsWithDistroId(distroID); err != nil {
+			return 0, InternalServerError.Send(ctx, fmt.Sprintf("decommissioning hosts for distro with id '%s': %s", distroID, err))
+		}
+		for _, h := range hosts {
+			event.LogHostStatusChanged(h.Id, h.Status, evergreen.HostDecommissioned, userID, "distro page")
+		}
+	case DistroOnSaveOperationReprovision:
+		catcher := grip.NewBasicCatcher()
+		for _, h := range hosts {
+			_, err = api.GetReprovisionToNewCallback(ctx, evergreen.GetEnvironment(), userID)(&h)
+			catcher.Wrapf(err, "marking host '%s' as needing to reprovision", h.Id, distroID)
+		}
+		if catcher.HasErrors() {
+			return 0, InternalServerError.Send(ctx, fmt.Sprintf("marking hosts for reprovision for distro with id '%s': %s", distroID, err))
+		}
+	case DistroOnSaveOperationRestartJasper:
+		catcher := grip.NewBasicCatcher()
+		for _, h := range hosts {
+			_, err = api.GetRestartJasperCallback(ctx, evergreen.GetEnvironment(), userID)(&h)
+			catcher.Wrapf(err, "marking host '%s' as needing Jasper service restarted", h.Id)
```

Comment:
Maybe we could just add the host ID to the catcher here since the error is wrapped in a similar message (and same above)